packages/synapse-react-client/src/components/TableColumnSchemaEditor/TableColumnSchemaForm.tsx

```diff
@@ -174,6 +177,9 @@ const TableColumnSchemaForm = React.forwardRef<
         submit() {
           onSubmit(readFormData())
         },
+        getEditedColumnModels() {
+          return transformFormDataToColumnModels(readFormData())
+        },
```

Comment:
The alternative to introducing this handle would probably be something like completely re-implementing the table/view creation wizard.